### PR TITLE
Remember rename settings

### LINKED ISSUE #56417: Inline-rename should remember the choices i made the last time i ran the feature.
I often do many renames in succession, using temp names and name swaps.  I often do not want to rename files as part of the rename i'm invoking (in fact, i never want this, but that's a digression).  When i rename i always have to disable this checkmark:

image

The action i've taken on the last rename should be remembered going forward.  It reflects a general decision on the part of hte user that we should use to feed into future choices.  I think it's fine for the default here to be checked, but that if hte user opts out we should respect that moving forward.

### PR DESCRIPTION
**Pull request by ryzngard**
Set rename settings as roaming storage so they are remembered for the user

Fixes #56417

### CODE REVIEW

File: src/Workspaces/Core/Portable/Rename/RenameOptions.cs [hunk 1]
```diff
@@ -10,16 +10,16 @@ namespace Microsoft.CodeAnalysis.Rename
...
-        public static Option<bool> RenameInStrings { get; } = new Option<bool>(nameof(RenameOptions), nameof(RenameInStrings), defaultValue: false);
-        public static Option<bool> RenameInComments { get; } = new Option<bool>(nameof(RenameOptions), nameof(RenameInComments), defaultValue: false);
+        public static Option<bool> RenameOverloads { get; } = new Option<bool>(nameof(RenameOptions), nameof(RenameOverloads), defaultValue: false, storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.RenameOverloads"));
+        public static Option<bool> RenameInStrings { get; } = new Option<bool>(nameof(RenameOptions), nameof(RenameInStrings), defaultValue: false, storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.RenameInStrings"));
+        public static Option<bool> RenameInComments { get; } = new Option<bool>(nameof(RenameOptions), nameof(RenameInComments), defaultValue: false, storageLocation: new RoamingProfileStorageLocation("TextEditor.%LANGUAGE%.Specific.RenameInComments"));
```
Comment (Youssef1313) [inline src/Workspaces/Core/Portable/Rename/RenameOptions.cs:15]: So my frequent review comment :smile: Does this need to be added to `AutomationObject`?

**Review by davidwengier — APPROVED** [review]

File: src/Workspaces/Core/Portable/Rename/RenameOptions.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Rename/RenameOptions.cs:15]: you're going to immediatley conflict with @tmat in this PR.  Pulling him in to clarify what we should be doing, and how it impacts his PR.

**Review by CyrusNajmabadi — CHANGES_REQUESTED** [review]
Please talk to tomas to make sure you're on the same page about what's going on here. (he's about to obsolete this type) :)

**Comment by tmat** [comment]
@ryzngard Would you mind if we make this change in 17.2?

**Comment by ryzngard** [comment]
We'll wait for option refactoring and do again in new architecture

**Comment by tmat** [comment]
I'll incorporate this to my change targeting 17.2.

**Comment by tmat** [comment]
Superseded by https://github.com/dotnet/roslyn/pull/58403

**Pull request closed without merging**